### PR TITLE
WIP: Compendium of Forgotten Secrets

### PR DESCRIPTION
**Pull request by ColdAnkles**
from Cold_Ankles:
Accursed Archive Patron
Fallen Exile Patron

from Azzy:
Ashen Lineage
Blackthorn Lineage
Spells
Races

Have some bits to do in the indexes, but making PR to allow review.

### CODE REVIEW

**Comment by Azzzzzy** [comment]
Heads up. I have my update section of the files to direct into ".../genuine-free-press/classes/..." and ".../genuine-free-press/races/..." so edit those according to how you want to organize the folders.

**Comment by ColdAnkles** [comment]
I settled on ".../genuine-free-press/forgotten-secrets/{races,archetypes}/..." for most things.

**Comment by swdriessen** [comment]
I'll need some time to look how the races work, they are not subraces, so probably will be racial traits. 

Also the selection of alternating ability score improvements will be racial trait selections since they do more than just increase 1 stat. I'll change and test it for one of the races as an example soon.

File: third-party/genuine-fantasy-press/forgotten-secrets/archetypes/barbarian-accursed-path.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    <rules>
+      <grant type="Spell" id="ID_GFP_COFSA_SPELL_DARK_SECRET" />
+    </rules>
+	</element>
+  
+  <element name="The Thing On The Doorstep" type="Archetype Feature" source="The Compendium of Forgotten Secrets" id="ID_GFP_COFSA_ARCHETYPE_PATH_OF_THE_ACCURSED_THE_THING_ON_THE_DOORSTEP">
+    <description>
+	    <p>At 10th level, your hallucinations slowly enter reality, as the dark tendrils of the Silent One move at the corners of your vision. Whenever you enter a rage, you can choose to unleash the horrors that dwell within your mind upon the world. Choose a number of creatures within 30 feet of you equal to or less than your Charisma modifier. Each one is pulled to an open space within 5 feet of you, or the nearest available space, by massive tendrils that spawn from the air around you. Until the end of your turn, when the tendrils disappear, you cannot suffer disadvantage from any source on any check made to grapple one of these creatures.</p>
+    </description>
+    <sheet usage="Start of Rage">
```
[the remark this answers is not in the document]
Comment (ColdAnkles) [reply third-party/genuine-fantasy-press/forgotten-secrets/archetypes/barbarian-accursed-path.xml]: Removed this usage as it's not amount restricted.

**Comment by swdriessen** [comment]
I've changed a bunch of things in the races

- changed racial origin from subrace to trait
- changed ASI selection to avoid confusion and display incorrect origin of stats and maintain display order from STR to CHA
- conditional spells from 'resistance from class feature' doesn't work as it was done, added optional selection as workaround to allow user to pick it if they meet the requirement
- and more (view commits for details)

I'll commit it later today if there are no pending changes to current files

**Comment by ColdAnkles** [comment]
Not planning on adding another warlock patron for another day (or two). Merge ahead! I'll probably open a part-2 pull request when I do though.

**Comment by swdriessen** [comment]
Thanks guys!

**Pull request merged**